src/context/history-context.js

```diff
@@ -0,0 +1,49 @@
+import { createContext, useContext, useReducer, useEffect } from "react";
+import { historyReducer } from "../reducer/historyReducer";
+import { useAuth } from "./auth-context";
+import {
+    addToHistoryHandler,
+    getHistoryHandler,
+    removeFromHistoryHandler,
+    resetHistoryHandler,
+    clearHistoryHandler,
+} from "../utils/history/history-utils";
+
+const defaultHistoryValue = {
+    itemInHistory: [],
+};
+
+const historyContext = createContext();
+
+const useHistory = () => useContext(historyContext);
+
+const HistoryProvider = ({ children }) => {
+    const [history, dispatchHistory] = useReducer(
+        historyReducer,
+        defaultHistoryValue
+    );
+
+    const { auth } = useAuth();
+
+    useEffect(() => {
+        auth.isAuthorized && getHistoryHandler(auth.token, dispatchHistory);
+    }, []);
+
+    return (
+        <historyContext.Provider
+            value={{
+                history,
+                dispatchHistory,
+                getHistoryHandler,
```

Comment:
Suggestion , sending your `utiles` in context is not needed , you can `import` your utiles their itself where you are using them

Reply:
I did this way because I directly want to import them with useHistory.